### PR TITLE
View Site: Enable modifier keys to behave expectedly on click

### LINKED ISSUE #17212: My Sites: clicking View Site with command no longer opens site in new tab
#### Steps to reproduce
1. Starting at URL: visit WordPress.com in a desktop browser logged into an account with at least one site
2. Click My Sites
3. Hold Command (Mac) on your keyboard and click My Sites
4. https://wordpress.com/view/example.com opens in a new tab instead of directly opening example.com

#### What I expected
example.com to open in a new tab

#### What happened instead
https://wordpress.com/view/example.com was opened in a new tab

#### Browser / OS version
Reproduced in Firefox 54.0, Safari 10.1.2, and Chrome 60.0.3112.90 on Macintosh OS X 10.11.6

#### Context / Source
#reallife #dogfooding

We'd also love to know how you found the bug: #dogfooding, #manual-testing

### PR DESCRIPTION
**Pull request by jblz**
Also, triggers GA `Sidebar` events for the various combinations of clicks on this menu item for tracking purposes:

* `Clicked View Site | Unpreviewable`
* `Clicked View Site | Modifier Key`
* `Clicked View Site | Calypso`

Fixes #17212

## To Test

* Hover over `View Site` -- it should show your site's URL
* Click on `View Site` in various situations for various types of sites (wpcom, jetpack, ssl, not) -- all previewable sites should open the section within calypso. Non-previewable sites should open in a new tab.
* Repeat the above with the cmd (or ctrl on non mac keyboards) key pressed. It should always open the site in a new tab. Other modifier keys behind pressed during clicking should behave as default / configured for your browser.
* Keyboard-based navigation (tab until `View Site` is "selected" & hit enter (with & without modifier keys being pressed) should all work according to browser defaults / preferences
* On mobile: long-press, open new tab should open the front end of the site in a new tab

### CODE REVIEW

**Comment by shaunandrews** [comment]
Safari makes this a little strange:

screen shot 2017-09-01 at 4 44 25 pm

**Comment by marekhrabe** [comment]
@shaunandrews That's what I talked about on Slack. There is no way we can assign one link two different URLs. Any solution with one link will always be a hack and we will break some contracts and expectations

**Comment by jblz** [comment]
> Safari makes this a little strange

@shaunandrews: Not sure I understand how that's strange. It's telling you what it'd do if you cmd+click, right?

**Comment by shaunandrews** [comment]
> Not sure I understand how that's strange. It's telling you what it'd do if you cmd+click, right?

Nope, thats the default tooltip when hovering the menu item — no modifier keys pressed.

**Comment by marekhrabe** [comment]
That's a right behavior, considering the link is `<a href="https://yoursite.com" target="_blank">View Site</a>`. That is indeed a link to yoursite.com that opens in a new tab. 

We let browsers do their job with it **unless** user used a left click **AND** no modified keys at all. In that case, we tell the browser to cancel the default click behavior (opening in a new tab) and we tell calypso to open `/view/yoursite.com` instead.

**Comment by jblz** [comment]
Ahh. I think we can pull out the `target` override when the site is "previewable". Will give it a shot.

**Comment by marekhrabe** [comment]
@jblz Just note that it will remove the `rel` which we need for security purposes

**Comment by jblz** [comment]
`target` & `rel` are now being omitted when we're not forcing these to be "internal" links
@shaunandrews Hover shouldn't show "in a tab" on Safari any longer.

@marekhrabe -- is this approach ok re: `rel`? It's there when the site is "unpreviewable."

**Comment by shaunandrews** [comment]
> That is indeed a link to yoursite.com that opens in a new tab.

I'm OK with that.

**Comment by marekhrabe** [comment]
I just checked the latest behavior:

- omitted `target="_blank"` is fine and brings no difference, except more logical link description in Safari 👍 
- omitted `rel=…` on the other hand opens a potential security hole. We need to keep it there for all external links, regardless of `target` or visual appearance

tldr: we need to bring `rel` back for all external links. After that, I think this PR will be good to go

**Comment by jblz** [comment]
Thanks for the review & adding the `rel` back in. 👍

**Comment by jblz** [comment]
@marekhrabe -- I tested this with a "real mouse" :) Middle click opens in a new tab as expected. 👍
